src/Pages/SignUp/SignUpInfo.js

```diff
@@ -0,0 +1,176 @@
+    super();
+
+    this.state = {
+      id: {
+        state: false,
+        value: "",
+        check: (value) => {
+          return /^[A-Za-z]{1}[A-Za-z0-9]{5,19}$/.test(value) ? true : false;
+        },
+      },
+      email: {
+        state: false,
+        value: "",
+        check: (value) => {
+          return /^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i.test(
+            value
+          )
+            ? true
+            : false;
+        },
+      },
+      password: {
+        state: false,
+        value: "",
+        check: (value) => {
+          return /^(?=.*[a-zA-Z])(?=.*[!@#$%^*+=-])(?=.*[0-9]).{8,19}$/.test(
+            value
+          )
+            ? true
+            : false;
+        },
+      },
+      passwordCheck: {
+        state: false,
+        value: "",
+        check: (value) => {
+          return this.state.password.value === value ? true : false;
+        },
+      },
+      userName: {
+        state: false,
+        value: "",
+        check: (value) => {
+          return /^[가-힣]{2,4}|[a-zA-Z]{2,10}\s[a-zA-Z]{2,10}$/.test(value)
+            ? true
+            : false;
+        },
+      },
+    };
+  }
+  validationCheck = (e) => {
+    const { name, value } = e.target;
+    const isValidation = this.state[name].check(value);
+    let prev = this.state[name];
+    prev.state = isValidation;
+    prev.value = value;
+    this.setState({ [name]: prev });
+  };
+  render() {
+    const { id, password, email, passwordCheck, userName } = this.state;
+    return (
+      <div className="SignUpInfo">
+        <StatusView status="info" />
+        <form>
+          <div>
+            <div>
+              <label>아이디</label>
+              <input
+                type="text"
+                name="id"
+                placeholder="아이디 입력"
+                onBlur={this.validationCheck}
+              />
+              <img
+                className={id.state ? "imgShow" : ""}
+                src="images/lock.png"
+                alt="id통과이미지"
+              />
+              <div className="errorMsg">
+                {id.value.length > 0
+                  ? id.state
+                    ? ""
+                    : "길이는 6~20 사이, 한글 및 특수문자는 사용하실수 없습니다."
+                  : ""}
+              </div>
+            </div>
+            <div>
+              <label>비밀번호</label>
+              <input
+                type="password"
+                name="password"
+                placeholder="비밀번호 입력"
+                onBlur={this.validationCheck}
+              />
+              <img
+                className={password.state ? "imgShow" : ""}
+                src="images/lock.png"
+                alt="비밀번호 패스 이미지"
+              />
+              <div className="errorMsg">
+                {password.value.length > 0
+                  ? password.state
+                    ? ""
+                    : "비밀번호는 최소 8자 이상, 특수문자, 대문자, 소문자를 입력해주세요"
+                  : ""}
```

Comment:
password.value.length 길이가 0보다 크다는 것은 1이상일때이죠? 
숫자가 0 이 아닌 1이상의 수의 boolean 속성을 무엇일까요? 그 속성을 이용해서 조건을 아래와 같이 작성해주실 수 있습니다.
```suggestion
                {password.value.length
                  ? password.state
                    ? ""
                    : "비밀번호는 최소 8자 이상, 특수문자, 대문자, 소문자를 입력해주세요"
                  : ""}
```

Reply:
맞아요 맞아요!! 잠시 잊고 있었내요